File: backend_old/project-managment-backend/src/main/java/at/htlleonding/entity/StudentClassHistory.java
package at.htlleonding.entity;

import jakarta.persistence.*;
import java.util.Objects;

@Entity
@Table(name = "student_class_history")
public class StudentClassHistory {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "history_id")
    public Long id;

    @ManyToOne(optional = false, fetch = FetchType.LAZY)
    @JoinColumn(name = "student_id", nullable = false)
    public Student student;

    @ManyToOne(optional = false, fetch = FetchType.LAZY)
    @JoinColumn(name = "class_id", nullable = false)
    public SchoolClass schoolClass;

    @ManyToOne(optional = false, fetch = FetchType.LAZY)
    @JoinColumn(name = "school_year_id", nullable = false)
    public SchoolYear schoolYear;

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof StudentClassHistory other)) return false;
        return id != null && Objects.equals(id, other.id);
    }

    @Override
    public int hashCode() {
        return 31;
    }
}

File: backend/WebApplication/Persistence/Entities/StudentClassHistory.cs
namespace Persistence.Entities;

public class StudentClassHistory : EntityObject
{
    public required string StudentId { get; set; }
    public Student? Student { get; set; }

    public int ClassId { get; set; }
    public StudentClass? StudentClass { get; set; }

    public int SchoolYearId { get; set; }
    public SchoolYear? SchoolYear { get; set; }

    public ICollection<ProjectStudent> ProjectStudents { get; init; } = [];
}
